packages/client-core/src/util/contextParams.ts

```diff
@@ -0,0 +1,42 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { LogParamsObject } from '@etherealengine/common/src/logger'
+
+/**
+ * @function clientContextParams
+ * @description This function will collect contextual parameters
+ * from url's query params
+ */
+export function clientContextParams(params: LogParamsObject) {
+  const urlParams = new URLSearchParams(window.location.search)
+
+  return {
+    ...params,
+    ...Object.fromEntries(urlParams),
```

Comment:
we should put tihs under a `search` property, as if a param is named `path` or `host` it will be overrided by the following values

Reply:
I see your point, though putting into a search property wouldn't address Balazs' requirements, so I guess I'll simply use the needed params, at the end he did mention that if a param is not available is better for him sending the param empty rather than not sending it at all, kind of { project_id: "" }. Will resubmit with these changes